### PR TITLE
paladin: latency monitoring and aggregation across modules

### PR DESCRIPTION
**Pull request by michael4338**

### CODE REVIEW

File: modules/monitor/software/latency_monitor.cc [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  if (current_time - flush_time_ > FLAGS_latency_report_interval) {
+    flush_time_ = current_time;
+    PublishLatencyReport();
+  }
+}
+
+void LatencyMonitor::UpdateLatencyStat(
+    const std::shared_ptr<LatencyRecordMap>& records) {
+  if (latency_report_.stat_aggr().find(records->module_name()) ==
```
Comment (xiaoxq) [inline modules/monitor/software/latency_monitor.cc]: Store `latency_report_.mutable_stat_aggr()` in a local pointer and reuse it. So the code is much simpler.

File: modules/monitor/software/latency_monitor.cc [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      latency_report_.stat_aggr().end()) {
+    LatencyStat stat;
+    latency_report_.mutable_stat_aggr()->insert({records->module_name(), stat});
```
Comment (xiaoxq) [inline modules/monitor/software/latency_monitor.cc]: Try `emplace`.

File: modules/monitor/software/latency_monitor.cc [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  }
+  auto* stat = &latency_report_.mutable_stat_aggr()->at(records->module_name());
+
+  uint64_t min_duration = (1UL << 63), max_duration = 0,
+           total_duration = 0;
+  for (const auto record : records->latency_records()) {
```
Comment (xiaoxq) [inline modules/monitor/software/latency_monitor.cc]: `auto&`

File: modules/monitor/software/latency_monitor.h [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+#include <vector>
+
+#include "modules/common/latency_recorder/proto/latency_record.pb.h"
+#include "modules/monitor/common/recurrent_runner.h"
+
+namespace apollo {
+namespace monitor {
+
+using apollo::common::LatencyRecordMap;
+using apollo::common::LatencyReport;
```
Comment (xiaoxq) [inline modules/monitor/software/latency_monitor.h]: Avoid making alias in header file. They introduced `apollo::monitor::LatencyReport` into the binary which is confusing and could cause problem.

File: modules/monitor/software/latency_monitor.cc (hunk 2 above)
Comment (michael4338) [reply modules/monitor/software/latency_monitor.cc]: the protobuf map does not seem to have "emplace" member function?

**Pull request merged**